### ISSUE TITLE
Total degree ignores the grading

### ISSUE THREAD

**Issue by simonbrandhorst** [oscar-system/Oscar.jl#2285]
**Describe the bug**
According to the documentation the total degree of a polynomial is the sum maximum degree of its monomial. For a non-standard graded ring the total degree ignores the grading. This is a bug or at least unexpected.

**To Reproduce**
```
using Oscar

julia> R,(x, y) = PolynomialRing(QQ,[:x,:y])
(Multivariate Polynomial Ring in x, y over Rational Field, QQMPolyRingElem[x, y])

julia> G,(x,y) = grade(R, [1, 2])
(Multivariate Polynomial Ring in x, y over Rational Field graded by 
  x -> [1]
  y -> [2], MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y])

julia> degree(x)
graded by [1]

julia> degree(y)
graded by [2]

julia> total_degree(y) # expected [2]
1 

julia> total_degree(y+x) # expected [2]
1 

```

**Comment by simonbrandhorst** [comment]
@wdecker

**Comment by thofma** [comment]
Since our groups are not ordered, what would the maximum be? We could so something magical, if the group happens to be `Z`?

**Comment by simonbrandhorst** [comment]
Good point. I had only a non-standard `Z`-Graded ring in mind.

**Comment by fieker** [comment]
On Tue, Apr 18, 2023 at 06:47:02AM -0700, Tommy Hofmann wrote:
> Since our groups are not ordered, what would the maximum be? We could so something magical, if the group happens to be `Z`?
> 
In the graded case all polynomials that have a degree are homogenous, so
total_degree = degree here...

**Comment by wdecker** [comment]
Yes,  we have total-degree referring to the usual total degree and degree referring to the degree of a homogeneous polynomial with respect to a given grading.

Von meinem iPhone gesendet

**Comment by simonbrandhorst** [comment]
"Usual degree" depends on what is usual in the area you work in. So at least it should be documented.

**Comment by fieker** [comment]
On Tue, Apr 18, 2023 at 08:14:53AM -0700, simonbrandhorst wrote:
>  "Usual degree" depends on what is usual in the area you work in. So at least it should be documented. 
> 
Thats why originally, the idea was to separate the graded and ungraded
world as much as possible: to avoid all misunderstandings and
cross-uses...

**Comment by simonbrandhorst** [comment]
Closed via #2290
